js/app/modules/selection/xapian.js

```diff
@@ -92,7 +92,8 @@ const Xapian = new Module.Class({
             }
 
             this._get_more = more;
-            let can_load_more = !!more && (info.upper_bound > results.length);
+            let can_load_more = !!more && (info.upper_bound > results.length ||
+                                           num_desired > results.length);
```

Comment:
I'm not sure I agree with this change, `can_load_more` should be independent of whether more results were requested.